app/services/publishers/vacancies/vacancy_step_process.rb

```diff
@@ -24,6 +24,14 @@ def validatable_steps
     steps - %i[documents review]
   end
 
+  def previous_visited_step
+    return steps.last if vacancy.completed_steps.include?(steps.last.to_s) && first_of_group?
+
+    return nil if current_step == steps.first
```

Comment:
Previous step should already return `nil` for this case so I don't think this line is necessary